x-pack/plugins/alerting/server/saved_objects/index.ts

```diff
@@ -150,12 +168,13 @@ export function setupSavedObjects(
   encryptedSavedObjects.registerType({
     type: RULE_SAVED_OBJECT_TYPE,
     attributesToEncrypt: new Set(['apiKey']),
-    attributesToExcludeFromAAD: new Set(RuleAttributesExcludedFromAAD),
+    attributesToIncludeInAAD: new Set(RuleAttributesIncludedInAAD),
   });
 
   // Encrypted attributes
   encryptedSavedObjects.registerType({
     type: 'api_key_pending_invalidation',
     attributesToEncrypt: new Set(['apiKeyId']),
+    attributesToIncludeInAAD: new Set('createdAt'),
```

Comment:
@elastic/response-ops I need some help with validating this object. I have manually created an instance from the main branch with all of the mapped properties, but it will not decrypt in this PR. I have tried removing the createdAt attribute from AAD as well, but this also does not decrypt. Any thoughts?

```
{
    "index": ".kibana_alerting_cases",
    "id": "api_key_pending_invalidation:bfda0e42-23d7-468c-8ed4-bf3c3de82cf3",
    "source": {
      "api_key_pending_invalidation": {
        "createdAt": "2024-01-23T03:25:33.002Z",
        "apiKeyId": "bhBpeJkCPf8xPNKpQMgIitKX9dFj+VDPiO8y/L0qv4M1w9f6InFW7PGa3aU84VhGGFWx/VSVvhlCg/GaH5BYH0YV3xMOjLImv2rus+strbJECjOWWgCqBAnthCx1bpEc8mvl7lLAZ2zJuhRPUatHEw6Sqmr6V/+V6nn7PhoVS7ceVdEzlAw2PwcT2Tt7poePob0fXeVu5ml3pU+RdvFtVQI="
      },
      "type": "api_key_pending_invalidation",
      "references": [],
      "managed": false,
      "coreMigrationVersion": "8.8.0",
      "updated_at": "2024-01-23T03:25:33.002Z",
      "created_at": "2024-01-23T03:25:33.002Z"
    }
  }
```

Reply:
I tested this the way it's intended to be used.  I'd be interested in finding out how you tested it, since the normal flow for these SO's happens at rule deletion time.

I set the following configs, to change the default values from 5m/1h to 30s/1m, for the API key invalidate task's interval and delay:

```
# how often the invalidate api keys task runs
xpack.alerting.invalidateApiKeysTask.interval: "30s" 

# how after a rule is deleted does the api key get removed
xpack.alerting.invalidateApiKeysTask.removalDelay: "1m"
```

Created a rule, deleted it.  Could see the `api_key_pending_invalidation` doc in the `.kibana_alerting_cases_*` index.  Could also see the API key in the API Key UX.  Waited for about a minute, then checked again - the `api_key_pending_invalidation` doc was deleted, as well as the API key.  No errors in the console.

So it looks like the code as-is is working as expected.

Reply:
@pmuellr What I am trying to validate is that the encryption definition for `api_key_pending_invalidation` objects is effectively unchanged from the previous implementation. The version of Kibnana running from this PR should be able to decrypt a `api_key_pending_invalidation` object that was encrypted by Kibana run from main.

There is a test for this object type here: https://github.com/elastic/kibana/blob/61fe1307eb80551ba174c6ef25227cfa6d5c48a0/x-pack/test/encrypted_saved_objects_api_integration/tests/encrypted_saved_objects_aad_include_list.ts#L84

I commented it out because it was failing to decrypt. It is concerning because the object is so simple. As far as I can tell there are only 2 attributes:  `createdAt` and `apiKeyId`. `apiKeyId` is encrypted, and `createdAt` should be included in the AAD. I've tried not including `createdAt` in the AAD, but this also fails.

Reply:
The `api_key_pending_invalidation` object I created from main is here: https://github.com/elastic/kibana/blob/61fe1307eb80551ba174c6ef25227cfa6d5c48a0/x-pack/test/encrypted_saved_objects_api_integration/fixtures/es_archiver/encrypted_saved_objects_pre_aad_change/data.json#L456

I just created this object manual from main by adding some code to an existing example plugin:
```
const objectsCreated = await savedObjects
    .getClient({
      includedHiddenTypes: ['api_key_pending_invalidation'],
      // excludedExtensions: [ENCRYPTION_EXTENSION_ID, SPACES_EXTENSION_ID],
    })
    .create(
      'api_key_pending_invalidation',
      {
        createdAt: new Date(), // '2024-01-17T22:46:44.927Z',
        apiKeyId: 'This api key id was encrypted prior to the AAD include list change.',
      }
```

Reply:
> Failed to decrypt attribute "apiKeyId" of saved object "api_key_pending_invalidation,bfda0e42-23d7-468c-8ed4-bf3c3de82cf3": Unsupported state or unable to authenticate data {"service":{"node":{"roles":["background_tasks","ui"]}}}

Reply:
Very strange!  I'm about to try testing a migration from 8.9.2 to this PR, to see what else might pop up, I'll look into this one after that ...

Reply:
For the migrations, I haven't tested the "token" type yet (for actions, I think for webhook oauth), but did test alert, action, action task params, and api key pending invalidation.  All seem to work fine but the last, which backups Jeramy's claim ^^^.  In my non-migration test yesterday, I don't believe I saw this.

Reply:
Here's the doc created in 8.9.2 that won't decrypt in this PR - figured I'd copy/post it before resetting my env:

<details>
<summary><code>api_key_pending_invalidation</code> SO that won't decrypt</summary>

```json
{
  "_index": ".kibana_alerting_cases_8.9.2_001",
  "_id": "api_key_pending_invalidation:18b595b0-c69e-11ee-90fe-616351eb9e2e",
  "_version": 1,
  "_score": 0,
  "_source": {
    "api_key_pending_invalidation": {
      "apiKeyId": "C9Af4XR4pRoVSQD5qPZVCEkYqqgowxss7/92aPGV5dqbwQCWZGsNXIcw9qs82We6J70CrQWXGrsvQZDyIODJASSTk8nE3bFf8f6u9BLN9ijorxR1Y5Z4ldwypK1iGFwjorlw0rjE5P28pCaOj6JCOnt6",
      "createdAt": "2024-02-08T16:21:21.674Z"
    },
    "type": "api_key_pending_invalidation",
    "references": [],
    "managed": false,
    "coreMigrationVersion": "8.8.0",
    "updated_at": "2024-02-08T16:21:21.675Z",
    "created_at": "2024-02-08T16:21:21.675Z"
  },
  "fields": {
    "api_key_pending_invalidation.createdAt": [
      "2024-02-08T16:21:21.674Z"
    ],
    "api_key_pending_invalidation.apiKeyId": [
      "C9Af4XR4pRoVSQD5qPZVCEkYqqgowxss7/92aPGV5dqbwQCWZGsNXIcw9qs82We6J70CrQWXGrsvQZDyIODJASSTk8nE3bFf8f6u9BLN9ijorxR1Y5Z4ldwypK1iGFwjorlw0rjE5P28pCaOj6JCOnt6"
    ],
    "updated_at": [
      "2024-02-08T16:21:21.675Z"
    ],
    "managed": [
      false
    ],
    "coreMigrationVersion": [
      "8.8.0"
    ],
    "created_at": [
      "2024-02-08T16:21:21.675Z"
    ],
    "type": [
      "api_key_pending_invalidation"
    ]
  }
}
```

</details>

Reply:
I found the issue! 😅 Missing brackets!
`attributesToIncludeInAAD: new Set('createdAt')`
vs.
`attributesToIncludeInAAD: new Set(['createdAt'])`

Reply:
OMG 😵‍💫 - I guess that created a set of 'c', 'r', 'e' ... !

I wonder if we should look for a lint rule for that - it's nasty!

Reply:
> OMG 😵‍💫 - I guess that created a set of 'c', 'r', 'e' ... !

Yes, and it was apparently very easily missed by human eyes (my own several times). I found it by adding some additional logging in the encrypted SO service. I don't think we would normally want to include the list of AAD attributes in our logs, but we could include some additional information when decryption fails - e.g. if the number of AAD attributes is zero. WDYT?

> I wonder if we should look for a lint rule for that - it's nasty!

Not a bad idea. I am not sure we ever intend to make a set from characters in Kibana anywhere - that could be verified first.

Reply:
I retested my migration from 8.9.2 to this PR, and the decryption now works!